### ISSUE TITLE
[BUG] - leadership-schedule not match with real situation 

### ISSUE THREAD

**Issue by eric248550** [input-output-hk/cardano-node#4583]
**Internal/External**
*External* 

**Area**
*Other*   Stake Pool

**Summary**
leadership-schedule not match with real situation 

**Steps to reproduce**
1. Run Leader logs command showed the poll was assigned the slot 75451711

2. When slot was 75451711 it shows `TraceNodeNotLeader `

**Expected behavior**
should match with leader-schedule

**System info (please complete the following information):**
- OS Name: amazon linux
- OS Version: amazon linux
- Node version: cardano-node 1.35.3 - linux-x86_64 - ghc-8.10
- CLI version : cardano-cli 1.35.3 - linux-x86_64 - ghc-8.10

**Screenshots and attachments**
**Run Leader logs command**
`cardano-cli query leadership-schedule --mainnet --genesis mainnet-shelley-genesis.json --stake-pool-id a59e2c8422ed40242742f3b6e8219fa0c454eaa7288d71981d3b65a9 --vrf-signing-key-file vrf.skey --current`
**Output**:
```
     SlotNo                          UTC Time              
-------------------------------------------------------------
     75451711                   2022-10-29 04:33:22 UTC
```
**When slot was 75451711**
```
Oct 29 04:33:22 cardano-node[6926]: [cardano.node.LeadershipCheck:Info:3453] [2022-10-29 04:33:22.00 UTC] {"chainDensity":4.892696e-2,"credentials":"Cardano","delegMapSize":1224583,"kind":"TraceStartLeadershipCheck","slot":75451711,"utxoSize":8834177}

Oct 29 04:33:22  cardano-node[6926]: [cardano.node.Forge:Info:3453] [2022-10-29 04:33:22.03 UTC] fromList [("credentials",String "Cardano"),("val",Object (fromList [("kind",String "TraceNodeNotLeader"),("slot",Number 7.5451711e7)]))]
```

**Comment by Jimbo4350** [comment]
Are you sure the logs belong to the same stake pool you are querying?

**Comment by eric248550** [comment]
> Are you sure the logs belong to the same stake pool you are querying?

The leader logs is from another server, but I used the vrf file from BP node.

**Comment by Robinyo** [comment]
We have also experienced the 'mystery slot' issue on 5 occasions since Cardano's Vasil hard fork.

For example:

```
     SlotNo                          UTC Time
-------------------------------------------------------------
     76216901                   2022-11-07 01:06:32 UTC
```

Core Node logs:

```
[f470fcd3:cardano.node.LeadershipCheck:Info:388] [2022-11-07 01:06:32.00 UTC] {"chainDensity":4.766215e-2,"credentials":"Cardano","delegMapSize":1226054,"kind":"TraceStartLeadershipCheck","slot":76216901,"utxoSize":8944413}
[f470fcd3:cardano.node.Forge:Info:388] [2022-11-07 01:06:32.06 UTC] fromList [("credentials",String "Cardano"),("val",Object (fromList [("kind",String "TraceNodeIsLeader"),("slot",Number 7.6216901e7)]))]
[f470fcd3:cardano.node.Forge:Info:388] [2022-11-07 01:06:32.06 UTC] fromList [("credentials",String "Cardano"),("val",Object (fromList [("block",String "c5cc647a6f60ff814b7adee7dbb2efc1b40dc9338faa6b99275a41ad013d1c1b"),("blockNo",Number 7981781.0),("blockPrev",String "d566ed7725334b2b1a36861dda8fa892d1e1dbe2cd293aa55b8c4b41f34b177d"),("kind",String "TraceForgedBlock"),("slot",Number 7.6216901e7)]))]
[f470fcd3:cardano.node.ChainDB:Notice:378] [2022-11-07 01:06:32.06 UTC] Chain extended, new tip: c5cc647a6f60ff814b7adee7dbb2efc1b40dc9338faa6b99275a41ad013d1c1b at slot 76216901
[f470fcd3:cardano.node.Forge:Info:388] [2022-11-07 01:06:32.06 UTC] fromList [("credentials",String "Cardano"),("val",Object (fromList [("blockHash",String "c5cc647a6f60ff814b7adee7dbb2efc1b40dc9338faa6b99275a41ad013d1c1b"),("blockSize",Number 871.0),("kind",String "TraceAdoptedBlock"),("slot",Number 7.6216901e7)]))]
```

<img width="512" alt="Screen Shot 2022-11-07 at 16 33 20" src="https://user-images.githubusercontent.com/12203540/200241547-54ba32eb-8b68-4d84-b563-accbae51ecb3.png">

**Comment by Robinyo** [comment]
See: https://cexplorer.io/block?query=76216901

**Comment by Robinyo** [comment]
First two were 'mystery slots', but minted the third one:

     SlotNo                          UTC Time
-------------------------------------------------------------
//   76216901                   2022-11-07 01:06:32 UTC
//   76336763                   2022-11-08 10:24:14 UTC
     76342462                   2022-11-08 11:59:13 UTC
     
see: https://cexplorer.io/block/98f0ca4a0850d314f2779493113ed3529e118e719b3b962380cfdda0354fd286

**Comment by disassembler** [comment]
@Robinyo Your issue is unrelated, since your logs show `NodeIsLeader` for that slot.

**Comment by CarlosLopezDeLara** [comment]
AFAIK this is solved here: https://github.com/input-output-hk/cardano-node/pull/4106 and is just waiting to be included in 1.36.0
